### DISCUSSION TITLE
Can't parse double slash "//"

### DISCUSSION THREAD

**Discussion by aimestereo** [alecthomas/participle#424] [Q&A] [answered]
Hello, I've took [simpleexpr](https://github.com/alecthomas/participle/blob/master/_examples/simpleexpr/main.go) example as my starting point and extended it to cover all my cases.

But for some reason I can't parse simple math operation: floor division: `//`

I've prepared a simplified example to highlight my exact case: 

Demo repo: https://github.com/aimestereo/go-dsl-floor-division-issue

```go
package main

import (
	"github.com/alecthomas/participle/v2"
)

type Expr struct {
    // doesn't matter if it's "/" "/" or "//"
	Op string `@(  "+" | "!" "=" | "&" "&" | "/" "/" | "/" )`
}

var (
	parser = participle.MustBuild[Expr]()
)
```

EBNF:
```ebnf
Expr = ("+" | ("!" "=") | ("&" "&") | ("/" "/") | "/") .
```

Diagram:
<img width="273" alt="image" src="https://github.com/user-attachments/assets/7976da6e-ee71-4e83-8448-47a515eb6102">

Tests that I'm checking:
* `+` - ok
* `!=` - ok
* `&&` - ok
* `/` - ok
* `//` - fails

**Comment by alecthomas** [comment, marked as answer]
You will need to define a custom lexer, the default lexer is the Go one which treats // as comments
